### ISSUE TITLE
Usability: Cancelling current tasks, and Manual cancelling

### ISSUE THREAD

**Issue by kevin-kline** [merico-dev/lake#775]
## Description
We currently have problems where the User can get stuck during a collection phase with no way to stop it. They can trigger multiple collections which can burn out their tokens, and it can be a very bad experience.

## Describe the solution you'd like

We need to implement the "Cancel Tasks" feature in 2 ways:
- [ ] We need the User to be able to manually cancel tasks through the Config-UI. 
- [x] We need to cancel current tasks upon re-trigger.

## Has the Feature been Requested Before?
No.

## Describe alternatives you've considered
None.

## Additional context
None.

**Comment by joncodo** [comment]
@e2corporation For the front end hook up, you just need to call `DEL localhost:8080/task/26`

**Comment by joncodo** [comment]
Backend work done. Assigning to @e2corporation

**Comment by klesh** [comment]
We had the `DELETE /task/:id` for cancelling task manually already.
And for automatical cancelling, we need to take requirement from EE into consideration.
I suggestion we do it after #824, and track it on #827 instead.

**Comment by joncodo** [comment]
No problem @klesh feel free to close my PR if you want.

**Comment by e2corporation** [comment]
Linked to #843, Deferred until revamp is done.

**Comment by hezyin** [comment]
This is waiting for #843, once the new design is out, we'll re-assign priority to this ticket.

**Comment by klesh** [comment]
Done
